### PR TITLE
Attempt to spell out history merger in more detail

### PR DESCRIPTION
**Pull request by domenic**
/cc @altimin @rakina @jakearchibald @noamr; not sure if you're the best reviewers but I thought you might be interested.

Before merging, we should get at least one an implementer review to make sure what I'm saying here seems reasonable from the point of view of our current implementation. @nyaxt maybe?

### CODE REVIEW

File: prerendering.bs [hunk 1]
```diff
@@ -429,7 +435,29 @@ Every {{Document}} has an <dfn for="Document">activation start time</dfn>, which
...
+
+       - Move |successorBC| from |successorTraversable| into |predecessorTraversable|.
+
+       - Move the session history entry |successorEntry| from |successorTraversable| into |predecessorTraversable|, as follows:
+
+          - If |historyHandling| is "`replace`", delete the currently-active session history entry of |predecessorTraversable|.
+
+          - If |historyHandling| is "`default`", delete every entry after the the currently-active session history entry of |predecessorTraversable|.
+
+          - Append |successorEntry| into the |predecessorTraversable|'s session history list, and make it the active session history entry. (As a side effect, this then makes |successorBC| the active browsing context of |predecessorTraversable|).
```
Comment (rakina) [inline prerendering.bs:456]: Shouldn't this try to replace the current session history in place instead of append if  |historyHandling| is "replace"?

**Review by nyaxt — APPROVED** [review]
LGTM myside

File: prerendering.bs (hunk 1 above)
Comment (domenic) [reply prerendering.bs:456]: If it's replace, we deleted the currently-active one, so appending at this point makes sense I think.

Comment (rakina) [reply prerendering.bs:456]: I was thinking of the case when the currently-active one is not the last entry in the session history list (e.g. after a back navigation?) but maybe that's OK too?

Comment (domenic) [reply prerendering.bs:456]: Oh, I think you're right, let me re-do it with that case in mind. Thank you!

**Pull request merged**